### PR TITLE
Feature cell fracs

### PR DESCRIPTION
**Pull request by zxkjack123**
Main changes:

- Add a method `tag_cell_fracs()` to `pyne/pyne/mesh.py:Mesh`. 

- Add a function `test_tag_cell_fracs()` in `pyne/tests/test_mesh.py` to test the adding method

Some white space in `pyne/tests/test_mesh.py` are deleted.

### CODE REVIEW

File: pyne/mesh.py [hunk 1]
```diff
@@ -1196,6 +1200,52 @@ def cell_fracs_to_mats(self, cell_fracs, cell_mats):
...
+        # set the cell_number_tag & cell_fracs_tag
+        self.cell_number_tag = IMeshTag(max_cell_number, int, mesh=self, name='cell_number_tag')
+        self.cell_fracs_tag = IMeshTag(max_cell_number, float, mesh=self,
+                                       name='cell_fracs_tag')
+
+        # fill the data
+        voxel_cell_number = np.empty(shape=(num_vol_elements,max_cell_number))
+        voxel_cell_fracs = np.empty(shape=(num_vol_elements,max_cell_number))
+        for ve in range(num_vol_elements):
+            voxel_cell_number[ve] = [int(-1)] * max_cell_number
```
Comment (scopatz) [inline pyne/mesh.py]: -1 is an integer, you don't need to call `int(-1)`

File: pyne/mesh.py [hunk 2]
```diff
@@ -1196,6 +1200,52 @@ def cell_fracs_to_mats(self, cell_fracs, cell_mats):
...
+            voxel_cell_fracs[ve] = [float(0.0)] * max_cell_number
```
Comment (scopatz) [inline pyne/mesh.py]: Similarly 0.0 is a float

File: pyne/r2s.py [hunk 3]
```diff
@@ -94,6 +94,10 @@ def irradiation_setup(flux_mesh, cell_mats, alara_params, tally_num=4,
 
     if m.structured:
         cell_fracs = discretize_geom(m, num_rays=num_rays, grid=grid)
+        # tag cell fracs
+        use_subvoxel = True
+        if use_subvoxel:
+            m.tag_cell_fracs(cell_fracs)
```
Comment (scopatz) [inline pyne/r2s.py]: These three lines are logically equivalent to only the last line, `m.tag_cell_fracs(cell_fracs)`

**Comment by scopatz** [comment]
A couple of minor suggestions @zxkjack123, but thanks for the PR!

**Comment by zxkjack123** [comment]
`mesh.py` and `r2s.py` are updated according to the suggestions of @scopatz .

**Comment by scopatz** [comment]
Thanks @zxkjack123 - I'll leave this for @gonuke, but it LGTM

File: pyne/mesh.py [hunk 4]
```diff
@@ -775,6 +775,14 @@ def __init__(self, mesh=None, structured=False,
                 setattr(self, name, MaterialMethodTag(mesh=self, name=name,
                         doc=doc))
 
+        # add a property: cell_cell_fracs_tag and set it to None
+        self.cell_number_tag = None
+        self.cell_fracs_tag = None
+        self.mat_name_tag = None
```
Comment (gonuke) [inline pyne/mesh.py:781]: Should take out the `mat_name_*` tags here since we aren't using them yet.

**Comment by gonuke** [comment]
Thanks @zxkjack123 - merging now.

File: pyne/mesh.py (hunk 4 above)
Comment (zxkjack123) [reply pyne/mesh.py:781]: I will delete them in next pull request.

**Pull request merged**